src/views/SecondAssessment.vue

```diff
@@ -0,0 +1,109 @@
+<script setup lang="ts">
+import { ref, onMounted } from 'vue'
+import { useRolesStore } from '@/stores/roles'
+import { ElSlider, ElButton } from 'element-plus'
+import type { Role } from '@/domain/Role'
+import type { Knowledge } from '@/domain/Knowledge'
+import { quizAnswer } from '@/stores/roles/state'
+import { ErrorMessage, SuccessMessage } from '@/services/messages'
+
+const roleId = ref('1')
+const rolesStore = useRolesStore()
+const role = ref<Role>({ name: '', knowledges: [], levels: [] })
+
+onMounted(async () => {
+  role.value = await rolesStore.getRoleById(roleId.value)
+  quizStateChange(role.value.knowledges ?? [])
+})
+
+const quizStateChange = (knowledges: Knowledge[]) => {
+  quizAnswer.value.answers = knowledges.map((knowledge) => ({
+    knowledgeId: Number(knowledge.id),
+    answer: '0'
+  }))
+}
+
+const getResult = async (quizAnswer: any) => {
+  if (quizAnswer) {
```

Comment:
That tip: check first the wrong case, and you won't need an else statement. As less else statements you have, the better!